### ISSUE TITLE
Tests fail

### ISSUE THREAD

**Issue by BastianZim** [pyscf/pyscf#1110]
Hi,

I'm currently in the process of adding the package to conda-forge and would have two questions regarding the test setup. I'm running the full test suite under _.github/workflows/run_tests.sh_ but am getting the following error:

OSX:

```shell
test_dhf (test_df_jk.KnownValues) ... .github/workflows/run_tests.sh: line 39: 21683 Segmentation fault: 11  nosetests pyscf/ -v --with-timer --with-cov --cov-report xml --cov-report term --cov-config .coveragerc --cov pyscf --exclude-dir=examples --exclude-dir=pyscf/pbc/grad -e test_bz -e h2o_vdz -e test_mc2step_4o4e -e test_ks_noimport -e test_jk_hermi0 -e test_j_kpts -e test_k_kpts -e high_cost -e skip -e call_in_background -e libxc_cam_beta_bug -e test_finite_diff_rks_eph -e test_finite_diff_uks_eph -e test_pipek -I test_kuccsd_supercell_vs_kpts\.py -I test_kccsd_ghf\.py -I test_h_.*\.py -I test_P_uadc_ea.py -I test_P_uadc_ip.py --exclude-test=pyscf/pbc/gw/test/test_kgw_slow_supercell.DiamondTestSupercell3 --exclude-test=pyscf/pbc/gw/test/test_kgw_slow_supercell.DiamondKSTestSupercell3 --exclude-test=pyscf/pbc/gw/test/test_kgw_slow.DiamondTestSupercell3 --exclude-test=pyscf/pbc/gw/test/test_kgw_slow.DiamondKSTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_krhf_slow_supercell.DiamondTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_kproxy_hf.DiamondTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_kproxy_ks.DiamondTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_kproxy_supercell_hf.DiamondTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_kproxy_supercell_ks.DiamondTestSupercell3 -I .*_slow.*py -I .*_kproxy_.*py -I test_proxy.py
```

Link: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=411933&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=7426

Linux: 

```shell
AttributeError: /home/conda/staged-recipes/build_artifacts/pyscf_1636984882810/test_tmp/pyscf/lib/libcc.so: undefined symbol: AO2MOnr_e2_drv
```

Link: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=411933&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823

Does anyone here know where this could come from?

Thanks!

**Comment by sunqm** [comment]
This is also observed in github actions. macos sometimes raises segfault error for a test in pbc.df module. But I couldn't reproduce the error on macos desktop

**Comment by BastianZim** [comment]
@sunqm Thanks for the info. I actually sort of figured that this just happens randomly which is why I closed this and opened #1118 for Windows only (Still working on that by the way). If you need more info or want me to run some tests, let me know.
